src/invidious/database/channels.cr

```diff
@@ -85,6 +85,25 @@ module Invidious::Database::Channels
 
     return PG_DB.query_all(request, ids, as: InvidiousChannel)
   end
+
+  # Select channels sorted by last published video
+  def select_sorted(ids : Array(String)) : Array(InvidiousChannel)?
+    return [] of InvidiousChannel if ids.empty?
+
+    request = <<-SQL
+      WITH max_published AS (
+        SELECT ucid, MAX(published) AS last_published
+        FROM channel_videos
+        GROUP BY ucid
+      )
```

Comment:
I'm not very familiar with SQL so please correct me if I'm wrong but doesn't this fetch every single row from the channels_videos column regardless of which channel IDs were requested? That seems inefficient

Reply:
Might be, it's been a while so I don't know what I was thinking (perhaps I looked it up and it was performant, or perhaps I thought I'll validate the idea first and optimize it after)
Either way, I think this will never merge, I'll just close the PR
Thanks for the review (and of course if this is useful in anyway please feel free to take over)